indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -151,7 +155,9 @@ public IndexTask(
       @JsonProperty("resource") TaskResource taskResource,
       @JsonProperty("spec") IndexIngestionSpec ingestionSchema,
       @JacksonInject ObjectMapper jsonMapper,
-      @JsonProperty("context") Map<String, Object> context
+      @JsonProperty("context") Map<String, Object> context,
+      @JacksonInject IndexMerger indexMerger,
+      @JacksonInject IndexIO indexIO
```

Comment:
@himanshug the toolbox shouldn't need to be injected just for serialization, as it's only needed if you're calling `run`. That wouldn't happen if the *Task class is just being used by something that wants to generate a request to submit to the overlord (like tranquility does). Right?

I am ok with doing it either way though. Tranquility already has to set up an ObjectMapper with a bunch of Druid modules (for stuff like AggregatorFactories and Firehoses) and so needing this other stuff too wouldn't be much worse.